installation_guide.md

```diff
@@ -0,0 +1,237 @@
+
+# PySyft Installation Guide
+
+Before you start following the tutorials, you need to install PySyft on your computer.
+
+Here are the PySyft installation instructions for Linux, MacOs and Windows.
+
+Just scroll down to find your preferred OS:
+
+--------------
+
+Authors:
+
+- Alan Aboudib - Twitter: [@alan_aboudib](https://twitter.com/alan_aboudib)
+
+
+
+----------------------------
+## 0.1 Linux
+--------------------
+
+Installing PySyft on Linux is really straight forward. Here are the steps:
+
+#### 1. Make sure you have python >= 3.6
+
+You can check out the version number by running:
+
+<font color='red'>$</font>  `python --version`
+
+#### 2. Install pytorch 1.0:
+
+Get the installation command here:
+
+https://pytorch.org/get-started/locally/. (use the pip option)
+
+and run it in the terminal.
+
+**<font color='red'>Attention:</font>** You might need to use `sudo` to run the installation command if you are not using Conda.
+
+
+#### 3. Clone the PySyft repo from github
+
+<font color='red'>$</font> `git clone https://github.com/OpenMined/PySyft.git`
+
+#### 4. Enter the cloned repo
+
+<font color='red'>$</font> `cd PySyft`
+
+#### 5. Install PySyft requirements
+
+<font color='red'>$</font> `sudo pip3 install -r requirements.txt`
+
+#### 6. Install PySyft
+
+<font color='red'>$</font> `sudo python setup.py install`
+
+#### 7. Test your installation:
+
+<font color='red'>$</font> `sudo python setup.py test`
+
+-----------------------
+## 0.2 Mac OS
+-------------------
+
+#### 1. Install python
+
+In order to install python on MacOs, you first need to install **Homebrew** the famous package manager.
+
+Start by opening a terminal and type the following:
+
+<font color='red'>$</font> `xcode-select --install`
+
+<font color='red'>$</font> `/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
+
+This will install **Homebrew**. Then, install python (version 3.6 or above) by running:
+
+<font color='red'>$</font> `brew install python3`
+
+#### 2. Install pytorch 1.0
+
+You should first get the installation command here:
+
+https://pytorch.org/get-started/locally/. (use the pip option)
+
+Then run that command in the open terminal.
+
+#### 3. Clone the PySyft repo from Github
+
+Just type
+
+<font color='red'>$</font> `git clone https://github.com/OpenMined/PySyft.git`
+
+In the terminal
+
+#### 4. Install PySyft
+
+First, you need to enter the PySyft folder you cloned earlier by typing:
+
+<font color='red'>$</font> `cd PySyft`
+
+in the terminal. Then, run this to install the PySyft package after installing its requirements:
+
+<font color='red'>$</font> `pip3 install -r requirements.txt`
```

Comment:
I know @iamtrask made a comment about this but this is not necessary. Setup.py handles this automatically